### PR TITLE
Implemented actions (WIP)

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/core/constants/pluginList.constant.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import {appVariables} from 'api/constants';
-import {PluginConfigInterface} from 'core/interfaces';
 import {PluginInterface} from 'core/interfaces/plugin.interface';
+import {CollaboarationPluginPropsInterface} from 'core/interfaces/pluginProps.interface';
 
 // This list later could be passed as parameters from API
-export const PLUGIN_LIST: (config: PluginConfigInterface) => PluginInterface[] = (config) => [
+export const COLLABORATION_PLUGIN_LIST: (
```
Comment (PAKovalev) [inline src/core/constants/pluginList.constant.ts:6]: Move this to store file.

File: src/core/interfaces/pluginProps.interface.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {AxiosInstance} from 'axios';
+
+import {ThemeInterface} from 'ui-kit';
+
+// TODO: Export it to a shared package so that plugins can use it.
+export interface PluginPropsInterface {
```
Comment (PAKovalev) [inline src/core/interfaces/pluginProps.interface.ts:6]: CorePluginPropsInterface

File: src/core/interfaces/pluginProps.interface.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  spaceId?: string;
+  isSpaceAdmin: boolean;
+  request?: AxiosInstance;
+}
+
+export interface PluginTopBarActionInterface {
+  main: () => JSX.Element | null;
+}
+
+export interface CollaboarationPluginPropsInterface extends PluginPropsInterface {
```
Comment (PAKovalev) [inline src/core/interfaces/pluginProps.interface.ts:17]: SpacePluginPropsInterface

File: src/core/interfaces/plugin.interface.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+import {PluginPropsInterface} from './pluginProps.interface';
 
 export interface PluginInterface {
   name: string;
   subPath: string;
   subtitle?: string;
   url: string;
   exact?: boolean;
-  config: PluginConfigInterface;
+  props?: PluginPropsInterface;
```
Comment (PAKovalev) [inline src/core/interfaces/plugin.interface.ts:9]: Remove this field.

File: src/scenes/collaboration/pages/CollaborationPluginPage/CollaborationPluginPage.tsx [hunk 5]
```diff
@@ -1,23 +1,31 @@
...
 import {useStore} from 'shared/hooks';
 import {SpaceTopBar, TextChat} from 'ui-kit';
+import {PluginLoader} from 'shared/hooks/pluginLoader';
+import {CollaborationPluginInterface} from 'scenes/collaboration/stores/CollaborationPluginsStore/models';
+import {PluginTopBarActionInterface} from 'core/interfaces';
 
 import * as styled from './CollaborationPluginPage.styled';
 
 interface PropsInterface {
-  subtitle?: string;
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/CollaborationPluginPage/CollaborationPluginPage.tsx:12]: Remove `PropsInterface`. Plugin key is in the url. Get plugin from store by key.

File: src/shared/hooks/pluginLoader.tsx [hunk 6]
```diff
@@ -17,10 +18,26 @@ const loadComponent = (scope: string, module: string) => async (): Promise<any>
...
+const PluginLoaderComponent: FC<PluginLoaderPropsInterface> = ({
+  name,
+  url,
+  props,
+  module = './App'
+}) => {
   const {ready, failed} = useDynamicScript(module && url);
   const {t} = useTranslation();
 
+  useEffect(() => {
```
Comment (PAKovalev) [inline src/shared/hooks/pluginLoader.tsx:37]: I feel we can remove this.

File: src/scenes/collaboration/Collaboration.tsx [hunk 7]
```diff
@@ -90,18 +88,9 @@ const Collaboration: FC = () => {
...
-        theme,
-        isSpaceAdmin: space?.isAdmin ?? false,
-        request,
-        spaceId
-      });
-
-      setPlugins(plugins);
-    }, 300);
-  }, [history, space?.isAdmin, spaceId, theme]);
+    pluginsStore.init(theme, space?.isAdmin ?? false, spaceId);
```
Comment (PAKovalev) [inline src/scenes/collaboration/Collaboration.tsx:91]: `pluginsStore.init(spaceId);`

File: src/scenes/collaboration/stores/CollaborationPluginsStore/models/CollaborationPlugin/CollaborationPlugin.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import {Instance, types} from 'mobx-state-tree';
+
+const CollaborationPlugin = types.model('CollaborationPlugin', {
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/CollaborationPluginsStore/models/CollaborationPlugin/CollaborationPlugin.ts:3]: SpacePlugin

**Review by PAKovalev — CHANGES_REQUESTED** [review]

File: src/scenes/collaboration/pages/CollaborationPluginPage/CollaborationPluginPage.tsx (hunk 5 above)
Comment (Daboo1998) [reply src/scenes/collaboration/pages/CollaborationPluginPage/CollaborationPluginPage.tsx:12]: But what about subtitle and request props?

File: src/shared/hooks/pluginLoader.tsx (hunk 6 above)
Comment (Daboo1998) [reply src/shared/hooks/pluginLoader.tsx:32]: it was for debugging, so yes we can now

File: src/scenes/collaboration/stores/CollaborationPluginsStore/CollaborationPluginsStore.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    types.model('CollaboarationPluginStore', {
+      plugins: types.array(CollaborationPlugin)
+    })
+  )
+  .actions((self) => ({
+    init(theme: ThemeInterface, isSpaceAdmin: boolean, spaceId: string) {
+      // TODO: Later change it to API call that returns this list
+      self.plugins = cast(
+        COLLABORATION_PLUGIN_LIST({
+          theme,
```
[the remark this answers is not in the document]
Comment (Daboo1998) [reply src/scenes/collaboration/stores/CollaborationPluginsStore/CollaborationPluginsStore.ts:21]: ah you mean they should be passed in component not store, I get it now

File: src/scenes/collaboration/Collaboration.tsx [hunk 10]
```diff
@@ -90,18 +88,9 @@ const Collaboration: FC = () => {
...
+    console.info('fetched plugin list', pluginsStore.plugins.length);
+  }, [history, pluginsStore, space?.isAdmin, spaceId, theme]);
```
[the remark this answers is not in the document]
Comment (Daboo1998) [reply src/scenes/collaboration/Collaboration.tsx:93]: I guess some leftover, Ill remove

File: src/scenes/collaboration/Collaboration.tsx [hunk 11]
```diff
@@ -200,7 +191,20 @@ const Collaboration: FC = () => {
...
+      <Switch>
+        {createRoutesByConfig(COLLABORATION_ROUTES)}
+        {pluginsStore.plugins.map((plugin) => {
+          return (
+            <Route
+              key={generatePath(ROUTES.collaboration.plugin, {subPath: plugin.subPath, spaceId})}
+              path={generatePath(ROUTES.collaboration.plugin, {subPath: plugin.subPath, spaceId})}
+              exact={plugin.exact}
+            >
+              <CollaborationPluginPage plugin={plugin} request={pluginRequest} />
```
[the remark this answers is not in the document]
Comment (Daboo1998) [reply src/scenes/collaboration/Collaboration.tsx]: But plugin needs some more information than just plugin name and path, like subtitle, isAdmin etc

File: src/scenes/collaboration/stores/CollaborationPluginsStore/models/CollaborationPlugin/CollaborationPlugin.ts (hunk 8 above)
Comment (Daboo1998) [reply src/scenes/collaboration/stores/CollaborationPluginsStore/models/CollaborationPlugin/CollaborationPlugin.ts:3]: I would agree but the plugin is in Collaboration so it more consistent to name it CollabprationPlugin

**Pull request closed without merging**